### PR TITLE
perf: reduce icons react bundle size

### PR DESCRIPTION
**Pull request by afc163**
## Summary

- split non-two-tone generated React icons onto a lighter runtime component
- keep two-tone color handling isolated to two-tone icons and public helpers
- avoid pulling the full `@rc-component/util` root entry into single-icon bundles
- add a bundle-size smoke script for representative React and SVG import patterns

## File diff overview

| File / group | What changed |
| --- | --- |
| `packages/icons-react/package.json` | Adds `test:bundle-size` to run the smoke bundle checks. |
| `packages/icons-react/scripts/bundle-size-smoke.js` | New esbuild-based smoke test covering direct ESM icon import, root named ESM import, SVG ASN import, and CJS compatibility import. |
| `packages/icons-react/scripts/generate.ts` | Generated React icons now choose runtime by theme: `TwoTone` keeps `AntdIcon`, all other themes use `AntdIconLight`. |
| `packages/icons-react/src/icons/*.tsx` | Generated import-only diff: non-two-tone icons import `../components/AntdIconLight`; two-tone icons continue to import `../components/AntdIcon`. Icon names, exports, and SVG definitions are unchanged. |
| `packages/icons-react/src/components/AntdIconLight.tsx` | New lightweight wrapper for non-two-tone icons, excluding two-tone color initialization and color context code from regular icon bundles. |
| `packages/icons-react/src/components/AntdIcon.tsx` | Stays as the full two-tone-capable wrapper and initializes the default two-tone color from the shared `@ant-design/colors` default instead of a hardcoded value. |
| `packages/icons-react/src/components/IconBase.tsx` | Reduced to plain object-icon rendering for non-two-tone icons. Function-based two-tone rendering moved out. |
| `packages/icons-react/src/components/IconBaseTwoTone.tsx` | New two-tone-specific base component containing the function icon path and primary/secondary color handling. |
| `packages/icons-react/src/components/twoTonePrimaryColor.ts`, `packages/icons-react/src/colorUtils.ts` | Moves two-tone color helpers into a focused module so non-two-tone bundles do not import color runtime. |
| `packages/icons-react/src/renderUtils.ts` | Hosts shared SVG render/style helpers used by icon components. |
| `packages/icons-react/src/cssUtils.ts`, `packages/icons-react/src/refUtils.ts` | Local minimal DOM style injection and ref composition helpers, avoiding the full `@rc-component/util` root entry in single-icon bundles. |
| `packages/icons-react/src/utils.ts` | Keeps the existing helper exports as compatibility re-exports; no public helper removal. |
| `packages/icons-react/src/index.ts` | Keeps `getTwoToneColor` and `setTwoToneColor` as runtime exports while exporting `TwoToneColor` through `export type`. |
| `packages/icons-svg/templates/*.ts`, `packages/icons-svg/**/*.ts` | Marks `IconDefinition` and related type-only imports as `import type`; generated SVG ASN output now emits type-only imports without runtime behavior changes. |

## Bundle size smoke

Overall across these representative smoke cases, total gzip size drops from **54.7 KB** to **25.6 KB**, saving **29.2 KB** (**-53.3%**). These numbers are the sum of the smoke scenarios, not package size. The biggest win is the CJS compatibility path, which drops from **19.5 KB** to **3.6 KB** gzip (**-81.5%**). The common direct ESM import of one outlined React icon drops from **11.3 KB** to **3.1 KB** gzip (**-72.6%**).

Before:

| Case | Bundle | Gzip |
| --- | ---: | ---: |
| react-one-outlined-direct-es | 26.4 KB | 11.3 KB |
| react-one-twotone-direct-es | 27.0 KB | 11.5 KB |
| react-three-root-named-es | 29.0 KB | 12.1 KB |
| svg-one-asn-direct-es | 0.9 KB | 0.5 KB |
| react-root-compat-cjs | 58.1 KB | 19.5 KB |

After:

| Case | Bundle | Gzip | Gzip saved |
| --- | ---: | ---: | ---: |
| react-one-outlined-direct-es | 6.9 KB | 3.1 KB | 8.2 KB (-72.6%) |
| react-one-twotone-direct-es | 20.9 KB | 8.8 KB | 2.7 KB (-23.3%) |
| react-three-root-named-es | 24.0 KB | 9.6 KB | 2.5 KB (-20.4%) |
| svg-one-asn-direct-es | 0.9 KB | 0.5 KB | 0 KB (no change) |
| react-root-compat-cjs | 10.3 KB | 3.6 KB | 15.9 KB (-81.5%) |

## Test Plan

- npm run test:bundle-size --workspace @ant-design/icons
- npm test --workspace @ant-design/icons -- Icon.test.tsx index.test.tsx type.test.tsx iconStyles.test.tsx
- npm run compile --workspace @ant-design/icons
- npm run icons:build
- npm run icons:test
- npm run react:ci

---

### 变更摘要

- 将非 TwoTone 的 React 图标切到更轻量的运行时组件 `AntdIconLight`。
- 将 TwoTone 的主色、次色和颜色计算逻辑隔离到 TwoTone 图标路径中。
- 避免单个图标 bundle 引入完整的 `@rc-component/util` 根入口。
- 新增 bundle-size smoke 脚本，覆盖 React 直引、React 根入口 named import、SVG ASN 直引和 CJS 兼容入口等代表性导入方式。

### 文件改动说明

| 文件 / 文件组 | 改动内容 |
| --- | --- |
| `packages/icons-react/package.json` | 新增 `test:bundle-size` 脚本，用于运行 bundle smoke 检查。 |
| `packages/icons-react/scripts/bundle-size-smoke.js` | 新增基于 esbuild 的 bundle 体积 smoke 测试。 |
| `packages/icons-react/scripts/generate.ts` | 生成 React 图标时按 theme 选择运行时：`TwoTone` 使用 `AntdIcon`，其他 theme 使用 `AntdIconLight`。 |
| `packages/icons-react/src/icons/*.tsx` | 生成文件的 import 变化：非 TwoTone 图标改为引入 `../components/AntdIconLight`；TwoTone 图标继续引入 `../components/AntdIcon`。图标名称、导出和 SVG 定义不变。 |
| `packages/icons-react/src/components/AntdIconLight.tsx` | 新增轻量运行时，只包含普通图标需要的 wrapper、className、spin、rotate、accessibility 和 SVG 渲染逻辑。 |
| `packages/icons-react/src/components/AntdIcon.tsx` | 保留完整 TwoTone 能力，并从共享的 `@ant-design/colors` 默认值初始化 TwoTone 默认主色，不写死颜色值。 |
| `packages/icons-react/src/components/IconBase.tsx` | 简化为普通对象图标渲染；TwoTone function icon 渲染逻辑移出。 |
| `packages/icons-react/src/components/IconBaseTwoTone.tsx` | 新增 TwoTone 专用基础组件，负责 function icon 路径和 primary / secondary color 处理。 |
| `packages/icons-react/src/components/twoTonePrimaryColor.ts`, `packages/icons-react/src/colorUtils.ts` | TwoTone 颜色 helper 集中到独立模块，避免非 TwoTone 图标 bundle 引入颜色运行时。 |
| `packages/icons-react/src/renderUtils.ts` | 承载图标组件共享的 SVG 渲染和样式注入 helper。 |
| `packages/icons-react/src/cssUtils.ts`, `packages/icons-react/src/refUtils.ts` | 本地化最小 DOM 样式注入和 ref 组合 helper，避免引入完整 `@rc-component/util` 根入口。 |
| `packages/icons-react/src/utils.ts` | 保留原有 helper 的兼容 re-export，不删除公开 helper。 |
| `packages/icons-react/src/index.ts` | 保持 `getTwoToneColor` 和 `setTwoToneColor` 为运行时导出，同时用 `export type` 导出 `TwoToneColor` 类型。 |
| `packages/icons-svg/templates/*.ts`, `packages/icons-svg/**/*.ts` | 将 `IconDefinition` 等纯类型导入改为 `import type`；生成的 SVG ASN 文件也会产出类型导入，运行时行为不变。 |

### Bundle Size 结果

以下数据是代表性 smoke case 的 bundle gzip 体积合计，不是 npm package size。合计 gzip 从 **54.7 KB** 降到 **25.6 KB**，节省 **29.2 KB**，降幅 **-53.3%**。

收益最大的场景是 CJS 兼容入口，gzip 从 **19.5 KB** 降到 **3.6 KB**，降幅 **-81.5%**。常见的单个 Outlined React 图标 ESM 直引，gzip 从 **11.3 KB** 降到 **3.1 KB**，降幅 **-72.6%**。

优化前：

| 场景 | Bundle | Gzip |
| --- | ---: | ---: |
| 单个 React Outlined ESM 直引 | 26.4 KB | 11.3 KB |
| 单个 React TwoTone ESM 直引 | 27.0 KB | 11.5 KB |
| React 根入口 named import 3 个图标 | 29.0 KB | 12.1 KB |
| 单个 `icons-svg` ASN ESM 直引 | 0.9 KB | 0.5 KB |
| React CJS 兼容入口 | 58.1 KB | 19.5 KB |

优化后：

| 场景 | Bundle | Gzip | Gzip 节省 |
| --- | ---: | ---: | ---: |
| 单个 React Outlined ESM 直引 | 6.9 KB | 3.1 KB | 8.2 KB (-72.6%) |
| 单个 React TwoTone ESM 直引 | 20.9 KB | 8.8 KB | 2.7 KB (-23.3%) |
| React 根入口 named import 3 个图标 | 24.0 KB | 9.6 KB | 2.5 KB (-20.4%) |
| 单个 `icons-svg` ASN ESM 直引 | 0.9 KB | 0.5 KB | 0 KB (无变化) |
| React CJS 兼容入口 | 10.3 KB | 3.6 KB | 15.9 KB (-81.5%) |

### 测试

- npm run test:bundle-size --workspace @ant-design/icons
- npm test --workspace @ant-design/icons -- Icon.test.tsx index.test.tsx type.test.tsx iconStyles.test.tsx
- npm run compile --workspace @ant-design/icons
- npm run icons:build
- npm run icons:test
- npm run react:ci

### CODE REVIEW

File: packages/icons-react/src/components/AntdIconLight.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    rotate,
+
+    tabIndex,
+    onClick,
+
+    // other
+    ...restProps
+  } = props;
+
+  delete restProps.twoToneColor;
```
[the remark this answers is not in the document]
Comment (afc163) [reply packages/icons-react/src/components/AntdIconLight.tsx]: Fixed by destructuring `twoToneColor` out before `restProps` in `AntdIconLight`, so the compatibility prop is ignored without mutating `restProps`.

File: packages/icons-react/src/components/IconBase.tsx [hunk 2]
```diff
@@ -12,50 +12,13 @@ export interface IconProps {
...
 
-  let colors: TwoToneColorPalette = twoToneColorPalette;
-  if (primaryColor) {
-    colors = {
-      primaryColor,
-      secondaryColor: secondaryColor || getSecondaryColor(primaryColor),
-    };
-  }
+  delete restProps.primaryColor;
+  delete restProps.secondaryColor;
```
[the remark this answers is not in the document]
Comment (afc163) [reply packages/icons-react/src/components/IconBase.tsx]: Fixed by destructuring `primaryColor` and `secondaryColor` out of `props` before collecting `restProps`, so these internal two-tone props are not forwarded and no `delete` operator is needed.

File: packages/icons-react/src/cssUtils.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  key: string,
+  originOption: {
+    attachTo?: Element | ShadowRoot;
+    csp?: { nonce?: string };
+    mark?: string;
+    prepend?: boolean | 'queue';
+    priority?: number;
+  } = {},
+) {
+  const container = getContainer(originOption);
```
[the remark this answers is not in the document]
Comment (afc163) [reply packages/icons-react/src/cssUtils.ts:173]: Fixed by adding an early `canUseDom()` guard in `updateCSS`, before any `document` access. Also added an SSR regression test that stubs `window` to `undefined` and verifies `updateCSS` returns `null` without throwing.
